class.c

```diff
@@ -965,17 +995,19 @@ rb_include_module(VALUE klass, VALUE module)
         rb_subclass_entry_t *iclass = RCLASS_SUBCLASSES(klass);
         int do_include = 1;
         while (iclass) {
-            VALUE check_class = iclass->klass;
-            while (check_class) {
-                if (RB_TYPE_P(check_class, T_ICLASS) &&
-                        (RBASIC(check_class)->klass == module)) {
-                    do_include = 0;
+            if (iclass->klass) {
+                VALUE check_class = iclass->klass;
+                while (check_class) {
+                    if (RB_TYPE_P(check_class, T_ICLASS) &&
+                            (RBASIC(check_class)->klass == module)) {
+                        do_include = 0;
+                    }
+                    check_class = RCLASS_SUPER(check_class);
                 }
-                check_class = RCLASS_SUPER(check_class);
-            }
 
-            if (do_include) {
-                include_modules_at(iclass->klass, RCLASS_ORIGIN(iclass->klass), module, TRUE);
+                if (do_include) {
+                    include_modules_at(iclass->klass, RCLASS_ORIGIN(iclass->klass), module, TRUE);
+                }
```

Comment:
Why is this change necessary?

Reply:
I added checks for the presence of `rb_subclass_entry->klass` into the places that walk the subclass list manually, as a way of effectively skipping the dummy head. My assumption is that having an `rb_subclass_entry` with no `klass` is not possible except for the dummy head.